File: webpack_src/data/Group.ts
export type Groups = {
    _embedded: {
        groups: Group[]
    }
}

export type Group = {
    id?: number,
    name: string,
    deleted?: boolean,
    subGroups?: string,
    _links?: {
        self: { href: string },
        group: { href: string },
        subGroups: { href: string }
    }
}

File: src/main/java/com/kehsa/spring_rest/database/entities/Group.java
package com.kehsa.spring_rest.database.entities;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "groups_")
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
@EqualsAndHashCode
public class Group {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    private boolean deleted;
    @OneToMany(mappedBy = "group")
    private List<SubGroup> subGroups;

    @PostLoad
    private void postLoad() {
        subGroups.removeIf(SubGroup::isDeleted);
    }

}